rfd/XXXX-tsh-upgrade.md

```diff
@@ -0,0 +1,117 @@
+---
+authors: Bernard Kim (bernard@goteleport.com)
+state: draft
+---
+# RFD XXXX - tsh upgrade
+
+## Required Approvers
+
+* Engineering: ...
+* Infra: ...
+* Kube: ...
+
+## What
+
+A tsh subcommand that will prompt the user to upgrade if the version of tsh is older than the cluster version.
+
+## Why
+
+One of our goals for Teleport Cloud is to increase the cadence at which clusters can be upgraded.
+This requires that Teleport Agents and Teleport client tools be up to date with the latest compatible version of the cluster.
+
+Manually upgrading Teleport Agents and client tools can be a tedious task, so customers can end up delaying their upgrades.
+Simplifying this process will help customers upgrade and stay up to date with their Teleport version.
+
+Development for Automatic Upgrades of Teleport Agents has already been completed, so the success criteria for this RFD only includes developing a simpler way to keep Teleport client tools up to date.
+
+## Details
+
+The main idea here is to add an upgrade command to the Teleport client tools. This will detect the version of the Teleport cluster.
+If the version of tsh is older than the version of the cluster, the user will be prompted to upgrade.
+The command should then re-download and re-install itself.
+
+### Identifying cluster version
+
+Detecting the version of the Teleport cluster should be straight forward.
+Teleport supports a ping RPC that returns a response containing information about the Teleport version of the cluster and the minimum client version required.
+In the case that the cluster version of Teleport is newer than the tsh version, the user will be prompted to upgrade.
+
+### Identifying installation method
+
+The Teleport client tools should be re-installed using the original installation method.
```

Comment:
We have discussed instead keeping a cache of tsh/tctl binaries of different versions somewhere on the client machine instead, right? Can you change this section to describe how that would work instead?